File: kattis/foundation/faktor/faktor.java
package kattis.faktor;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class faktor {

    private static final boolean debug = false;

    public static void main(String[] args) {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        try {
            // Take the line input
            String inputJournalInfo = bufferedReader.readLine();
            if (inputJournalInfo == null || inputJournalInfo.trim().isEmpty()) {
                dPrint("Input error");
                return;
            }

            // Start the timer
            long startRuntime = System.nanoTime();

            // Find journal info
            String[] journalInfo = inputJournalInfo.trim().split(" ");
            if (journalInfo.length != 2) {
                dPrint("Wrong journal info");
                return;
            }

            int numberOfArticle = Integer.parseInt(journalInfo[0].trim());
            int impactFactor = Integer.parseInt(journalInfo[1].trim());
            dPrint("Number of Article: " + numberOfArticle + " and impact factor " + impactFactor);

            if (!isValidJournalInfo(numberOfArticle) || !isValidJournalInfo(impactFactor)) {
                dPrint("Invalid journal info");
                return;
            }

            // Mathematical calculation matching your formula
            int bribedScientist = numberOfArticle * (impactFactor - 1) + 1;
            System.out.println(bribedScientist);

            // Stop the timer here
            long endRuntime = System.nanoTime();
            double totalRuntime = (endRuntime - startRuntime) / 1_000_000_000.0;

            dPrint(String.format("[Algorithmic Runtime] %.6f seconds", totalRuntime));

        } catch (IOException e) {
            dPrint("IOException occurred: " + e.getMessage());
        }
    }

    private static void dPrint(Object... args) {
        if (debug) {
            StringBuilder sb = new StringBuilder();
            for (Object arg : args) {
                sb.append(arg).append(" ");
            }
            System.out.println(sb.toString().trim());
        }
    }

    private static boolean isValidJournalInfo(int metadata) {
        int min = 1;
        int max = 100;
        return metadata >= min && metadata <= max;
    }
}

File: kattis/foundation/faktor/faktor.py
import sys
import time


class faktor:

    debug = False

    @staticmethod
    def dPrint(*args, **kwargs):
        if faktor.debug:
            print(*args, **kwargs)

    @staticmethod
    def isValidJournalInfo(metadata: int) -> bool:
        min = 1
        max = 100
        return min <= metadata <= max

    @staticmethod
    def generateOutput():
        # Take the line input
        inputJournalInfo = sys.stdin.readline().strip()
        if not input:
            faktor.dPrint("Input error")
            return

        # Start the timer
        startRuntime = time.perf_counter()

        # Find journal info
        journalInfo = inputJournalInfo.split()
        if len(journalInfo) != 2:
            faktor.dPrint("Wrong journal info")
            return

        numberOfArticle = int(journalInfo[0].strip())
        impactFactor = int(journalInfo[1].strip())
        faktor.dPrint(
            f"Number of Article: {numberOfArticle} and impact factor {impactFactor}"
        )

        if any(
            not faktor.isValidJournalInfo(i) for i in [numberOfArticle, impactFactor]
        ):
            faktor.dPrint("Invalid journal info")

        bribedScientist = numberOfArticle * (impactFactor - 1) + 1
        print(bribedScientist)

        # Stop the timer here
        endRuntime = time.perf_counter()
        totalRuntime = endRuntime - startRuntime

        faktor.dPrint(f"[Algorithmic Runtime] {totalRuntime:.6f} seconds")


if __name__ == "__main__":
    faktor.generateOutput()
